fitbenchmarking/core/tests/test_fitting_benchmarking.py

```diff
@@ -0,0 +1,824 @@
+        """
+        data_file = DATA_DIR + 'ENSO.dat'
+
+        options = Options(additional_options={'software': ['scipy'],
+                                              'table_type': ['acc',
+                                                             'runtime',
+                                                             'compare',
+                                                             'local_min']})
+        cp = Checkpoint(options)
+
+        self.parsed_problem = parse_problem_file(data_file, options)
+        self.parsed_problem.correct_data()
+
+        self.fit = Fit(options=options,
+                       data_dir=data_file,
+                       checkpointer=cp)
+
+    @patch(f"{FITTING_DIR}.Fit._Fit__loop_over_cost_function",
+           side_effect=mock_loop_over_cost_func_call)
+    def test_loop_over_starting_values(self, mock):
+        """
+        The test checks __loop_over_starting_values method.
+        Enso.dat /NIST/average_difficulty problem set is used.
+        """
+        results = self.fit._Fit__loop_over_starting_values(self.parsed_problem)
+
+        assert len(results) == 18
+        assert mock.call_count == 2
+        assert all(isinstance(r, FittingResult) for r in results)
+
+    @patch(f"{FITTING_DIR}.Fit._Fit__loop_over_cost_function",
+           side_effect=mock_loop_over_cost_func_call_all_fail)
+    def test_loop_over_starting_values_fail(self, mock):
+        """
+        The test checks that _failed_problems is updated correctly.
+        """
+        _ = self.fit._Fit__loop_over_starting_values(self.parsed_problem)
+
+        assert self.fit._failed_problems == ['ENSO, Start 1', 'ENSO, Start 2']
+        assert mock.call_count == 2
+
+
+class BenchmarkTests(unittest.TestCase):
+    """
+    Verifies the output of the benchmarking method in the Fit class
+    when run with different options.
+    """
+
+    def setUp(self):
+        """
+        Initializes the fit class for the tests
+        """
+        options = Options(additional_options={'software': ['scipy_ls'],
+                                              'table_type': ['acc',
+                                                             'runtime',
+                                                             'compare',
+                                                             'local_min']})
+        cp = Checkpoint(options)
+        self.fit = Fit(options=options,
+                       data_dir=DATA_DIR,
+                       checkpointer=cp)
+
+    def test_benchmark_method(self):
+        """
+        This regression test checks fitbenchmarking with the default
+        software options.The results of running the new class are matched
+        with the orginal benchmark function. The /NIST/average_difficulty
+        problem set is run with scipy_ls software.
+        """
+        results_dir = ROOT + \
+            "/fitbenchmarking/test_files/regression_checkpoint.json"
```

Comment:
```suggestion
        results_dir = os.path.join(TEST_FILES_DIR, "regression_checkpoint.json")
```
We could change the name of  "regression_checkpoint.json" to something that includes the name of the test_file, something like  "fitting_benchmarking_regression_cp.json" or so